nodeadm/internal/kubelet/environment.go

```diff
@@ -13,21 +13,28 @@ const (
 	kubeletArgsEnvironmentName = "NODEADM_KUBELET_ARGS"
 )
 
-// Write environment variables for kubelet execution
+// Write environment variables needed for kubelet runtime. This should be the
+// last method called on the kubelet object so that environment side effects of
+// other methods are properly recored
 func (k *kubelet) writeKubeletEnvironment(cfg *api.NodeConfig) error {
-	// transform kubelet additional arguments into a string and write them to
-	// the kubelet args environment variable
-	kubeletArgs := make([]string, len(k.additionalArguments))
-	for flag, value := range k.additionalArguments {
-		kubeletArgs = append(kubeletArgs, fmt.Sprintf("--%s=%s", flag, value))
+	// overwrite the kubelet cli arguments with flags specified by the user
+	for flag, flagValue := range cfg.Spec.Kubelet.Flags {
+		k.additionalArguments[flag] = flagValue
 	}
-	kubeletEnvironment := []string{
-		fmt.Sprintf("%s=%s", kubeletArgsEnvironmentName, strings.Join(kubeletArgs, " ")),
+	// transform kubelet arguments into a string and write them to the kubelet
+	// environment variable
+	var kubeletFlags []string
+	for flag, value := range k.additionalArguments {
+		kubeletFlags = append(kubeletFlags, fmt.Sprintf("--%s=%s", flag, value))
 	}
+
+	var kubeletEnvironment []string
+	kubeletEnvironment = append(kubeletEnvironment, fmt.Sprintf("%s=%s", kubeletArgsEnvironmentName, strings.Join(kubeletFlags, " ")))
```

Comment:
Does the value of this environment variable need to be quoted?

i.e.
```
KUBELET_ARGS=--foo --bar
```
or
```
KUBELET_ARGS='--foo --bar'
```

Reply:
didnt see any hard requirements for it to be quoted, let me triple check

Reply:
going to fix this to double quotes per https://github.com/markstos/systemd/blob/master/man/systemd.exec.xml#L345-L363